src/renderers/native-rt/ReactNativeRTFiberRenderer.js

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @providesModule ReactNativeRTFiberRenderer
+ * @flow
+ */
+
+'use strict';
+
+const ReactFiberReconciler = require('ReactFiberReconciler');
+const ReactNativeRTComponentTree = require('ReactNativeRTComponentTree');
+const ReactNativeRTTagHandles = require('ReactNativeRTTagHandles');
+const RTManager = require('RTManager');
+
+const emptyObject = require('fbjs/lib/emptyObject');
+const invariant = require('fbjs/lib/invariant');
+
+export type Container = number;
+export type Instance = number;
+export type Props = Object;
+export type TextInstance = number;
+
+const {precacheFiberNode, updateFiberProps} = ReactNativeRTComponentTree;
+
+const NativeRTRenderer = ReactFiberReconciler({
+  appendChild(parentInstance: Instance, child: Instance | TextInstance): void {
+    RTManager.appendChild(parentInstance, child);
+  },
+
+  appendChildToContainer(
+    parentInstance: Container,
+    child: Instance | TextInstance,
+  ): void {
+    RTManager.appendChild(parentInstance, child);
+  },
+
+  appendInitialChild(
+    parentInstance: Instance,
+    child: Instance | TextInstance,
+  ): void {
+    RTManager.appendChildToDetachedParent(parentInstance, child);
+  },
+
+  commitTextUpdate(
+    textInstance: TextInstance,
+    oldText: string,
+    newText: string,
+  ): void {
+    invariant(false, 'Text components are not yet supported.');
+  },
+
+  commitMount(
+    instance: Instance,
+    type: string,
+    newProps: Props,
+    internalInstanceHandle: Object,
+  ): void {
+    // Noop
+  },
+
+  commitUpdate(
+    instance: Instance,
+    updatePayload: Object,
+    type: string,
+    oldProps: Props,
+    newProps: Props,
+    internalInstanceHandle: Object,
+  ): void {
+    updateFiberProps(instance, newProps);
+    RTManager.updateNode(instance, newProps);
+  },
+
+  createInstance(
+    type: string,
+    props: Props,
+    rootContainerInstance: Container,
+    hostContext: {},
+    internalInstanceHandle: Object,
+  ): Instance {
+    const tag = ReactNativeRTTagHandles.allocateTag();
+    RTManager.createNode(tag, type, props);
+    precacheFiberNode(internalInstanceHandle, tag);
+    updateFiberProps(tag, props);
```

Comment:
We update props in `commitUpdate` and `createInstance`. Is that odd? Should it be  in `commitUpdate` and `commitMount` instead? (I realize you're doing the same thing React Native does 
 but I'm questioning that now too.)

Reply:
That is updating the internal cache slot that keeps the "current event listeners". Really should only need to update when event listeners change but it's cheaper to just update it than checking.

Reply:
My question was more about the fact that we update this during _commit_ phase in one place and _complete_ phase in another. I was asking if we should wait to update in _commit_ phase in both places.

Reply:
Right. We also do that with any other props since it's better to start work during the render phase if we can do as much work as possible prepared before we commit so that the commit is fast. During the initial mount phase nothing is in the "tree" yet so we won't get any events.

The only quirk here is that this set will leak if it doesn't commit, which is a more general problem.

Reply:
> The only quirk here is that this set will leak if it doesn't commit, which is a more general problem.

Thanks for clarifying. This is more what I was concerned about ^ but if this is a conscious decision on your part, then I'll defer to your judgement. 😄